gratipay/models/participant/__init__.py

```diff
@@ -1135,8 +1135,11 @@ def change_username(self, suggested):
 
         lowercased = suggested.lower()
 
-        if lowercased in gratipay.RESTRICTED_USERNAMES:
-            raise UsernameIsRestricted(suggested)
+        # Don't allow any username which is the name of a
+        # file existing on the web_root folder.
+        for ext in ['', '.spt', '.ico', '.txt', '.osdd']:
+            if (lowercased + ext) in gratipay.RESTRICTED_USERNAMES:
+                raise UsernameIsRestricted(suggested)
```

Comment:
That's basically an internal redirect, which we don't have good facilities for in Aspen (https://github.com/AspenWeb/pando.py/issues/143). The test harness is the closest we have, and we don't want to import all that machinery into production. Hmm ...